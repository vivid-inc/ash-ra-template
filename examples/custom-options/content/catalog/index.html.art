{%

(require '[hiccup.core :as hiccup])
(require '[com.acme.ordering])

%}

<h1>{{manufacturer}} Product Catalog {{manufacture-year}}</h1>

<!-- hiccup employed only for tabular content -->
<h2>Sales Offices</h2>
<p>Contraption design &amp; consulting services available through all sales offices.</p>
<table>
  <thead>
    <tr>
      <th>Region</th>
      <th>Phone Number</th>
    </tr>
  </thead>
  <tbody>
    {{ (hiccup/html
         (for [{:keys [:region :phone]} sales-offices]
           (hiccup/html [:tr [:td region] [:td phone]]))) }}
  </tbody>
</table>
<p>Or contact any of our eager partners at {{ (clojure.string/join " or " partner-list) }}.</p>

<!-- HTML generated completely with hiccup -->
{{
(hiccup/html
  [:h2 "Products"
    [:table
      [:thead [:tr [:th "Name"] [:th "Weight"] [:th "Minimum Order Qty"]]]
      [:tbody
        (hiccup/html
          (for [{:keys [name weight-kgs minimum-order-qty unit-price-dollars]} products]
             (hiccup/html [:tr [:td name]
                               [:td (str weight-kgs " kg")]
                               [:td (com.acme.ordering/minimum-order-qty unit-price-dollars)]
                               [:td (str unit-price-dollars " $")]])))]]])
}}

<footer>
  Copyright (c) {{ current-year }} {{ manufacturer }}
</footer>
