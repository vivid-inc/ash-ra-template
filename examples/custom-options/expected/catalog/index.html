

<h1>Acme Corporation Product Catalog 2022</h1>

<!-- hiccup employed only for tabular content -->
<h2>Sales Offices</h2>
<p>Contraption design &amp; consulting services available through all sales offices.</p>
<table>
  <thead>
    <tr>
      <th>Region</th>
      <th>Phone Number</th>
    </tr>
  </thead>
  <tbody>
    <tr><td>Sonoran</td><td>+1-800-123-4567</td></tr><tr><td>Mojave</td><td>+7654-321-008-1</td></tr>
  </tbody>
</table>

<!-- HTML generated completely with hiccup -->
<h2>Products<table><thead><tr><th>Name</th><th>Weight</th><th>Minimum Order Qty</th></tr></thead><tbody><tr><td>Bag of bird seed</td><td>1.0 kg</td><td>129</td><td>0.39 $</td></tr><tr><td>Ironing board on rollerskates</td><td>2.0 kg</td><td>3</td><td>17.95 $</td></tr></tbody></table></h2>

<footer>
  Copyright (c) 2021 Acme Corporation
</footer>
